src/balances/balances.controller.spec.ts

```diff
@@ -19,95 +23,282 @@ describe('Balances Controller (Unit)', () => {
-          fiatConversion: 2,
-        },
-      ],
-    };
-
-    const exchangeResponse = {
-      data: {
-        success: true,
-        timestamp: 1660567445,
-        base: 'EUR',
-        date: '2022-08-15',
-        rates: {
-          USD: 2,
-        },
-      },
-    };
-
-    const safeConfigServiceResponse = {
-      data: {
-        chainId: '1',
-        chainName: 'mainnet',
-        transactionService: 'https://test.safe.transaction.service',
-        vpcTransactionService: 'https://test.safe.transaction.service',
-      },
-    };
-    axiosMock.get.mockImplementation((url) => {
-      if (url == 'https://safe-config.gnosis.io/api/v1/chains/1') {
-        return Promise.resolve(safeConfigServiceResponse);
-      } else if (
-        url ==
-        `https://test.safe.transaction.service/api/v1/safes/0x0000000000000000000000000000000000000001/balances/usd/`
-      ) {
-        return Promise.resolve(safeTransactionServiceBalancesResponse);
-      } else if (url == 'http://api.exchangeratesapi.io/latest') {
-        return Promise.resolve(exchangeResponse);
-      } else {
-        return Promise.reject(new Error(`Could not match ${url}`));
-      }
-    });
+  afterAll(async () => {
+    await app.close();
+  });
 
-    await request(app.getHttpServer())
-      .get(`/chains/${chainId}/safes/${safeAddress}/balances/USD`)
-      .expect(200)
-      .expect({
-        fiatTotal: 100,
-        items: [
-          {
-            tokenInfo: {
-              tokenType: 'ERC20',
-              address: '0x6810e776880C02933D47DB1b9fc05908e5386b96',
-              decimals: 16,
-              symbol: 'GNO',
-              name: 'Gnosis',
-              logoUri: null,
+  describe('GET /balances', () => {
+    it(`Success`, async () => {
+      const chainId = '1';
+      const safeAddress = '0x0000000000000000000000000000000000000001';
+      const safeTransactionServiceBalancesResponse = safeBalanceFactory(1);
+      const exchangeResponse = exchangeResultFactory({ USD: 2.0 });
+      const chainResponse = chainFactory(chainId);
+      axiosMock.get.mockImplementation((url) => {
+        if (url == `https://safe-config.gnosis.io/api/v1/chains/${chainId}`) {
+          return Promise.resolve({ data: chainResponse });
+        } else if (
+          url ==
+          `${chainResponse.transactionService}/api/v1/safes/${safeAddress}/balances/usd/`
+        ) {
+          return Promise.resolve({
+            data: safeTransactionServiceBalancesResponse,
+          });
+        } else if (url == configuration().exchange.baseUri) {
+          return Promise.resolve({ data: exchangeResponse });
+        } else {
+          return Promise.reject(new Error(`Could not match ${url}`));
+        }
+      });
+
+      await request(app.getHttpServer())
+        .get(`/chains/${chainId}/safes/${safeAddress}/balances/usd`)
+        .expect(200)
+        .expect({
+          fiatTotal: safeTransactionServiceBalancesResponse[0].fiatBalance,
```

Comment:
NP: maybe if you extract the first element in the array beforehand it could be (in my opinion) a bit more readable? What do you think? 🙂
![Screenshot 2022-08-17 at 09 38 58](https://user-images.githubusercontent.com/21182069/185062189-5fe2a2e7-afe0-4b3f-a579-297994df1b91.png)

Reply:
[ca8b3d8](https://github.com/5afe/safe-client-gateway-nest/pull/31/commits/ca8b3d84b29e134fddb0f2fdb917ce363230b7d1)